code/__HELPERS/unsorted.dm

```diff
@@ -1259,3 +1259,42 @@ will handle it, but:
 		pois[name] = M
 
 	return pois
+
+//Converts input direction to a list of either two same directions if the input is cardinal
+//Otherwise will return the two directions that make up the diagonal
+/proc/DiagonalToCardinal(direct)
+	if(direct in GLOB.cardinals)
+		return list(direct, direct)
+	return list(NSCOMPONENT(direct), EWCOMPONENT(direct))
+
+	switch(direct)
```

Comment:
unreachable code

Reply:
Diagonal to cardinal proc no longer needed